core/conversion/converters/impl/plugins/adaptive_max_pool2d_plugin.cpp

```diff
@@ -0,0 +1,317 @@
+  output_archive.write("size", torch::IValue(size_));
+  output_archive.write("mode", torch::IValue(mode_));
+  output_archive.write("align_corners", torch::IValue(align_corners_));
+
+  std::ostringstream data_str;
+  output_archive.save_to(data_str);
+
+  return data_str.str();
+}
+
+size_t AdaptiveMaxPool2dPlugin::getSerializationSize() const {
+  return serializeToString().size();
+}
+
+bool AdaptiveMaxPool2dPlugin::supportsFormatCombination(
+    int pos,
+    const nvinfer1::PluginTensorDesc* inOut,
+    int nbInputs,
+    int nbOutputs) {
+  TRTORCH_ASSERT(0 <= pos && pos <= 2, "There should be exactly 3 connections to the plugin - 1 input, 2 output");
+  TRTORCH_ASSERT(nbInputs == 1, "Expected a single tensor as input to interpolate plugin");
+  TRTORCH_ASSERT(nbOutputs == 2, "Expected two tensors as output to interpolate plugin");
+
+  const PluginTensorDesc& in = inOut[0];
+
+  if (pos == 0) {
+    return (in.type == nvinfer1::DataType::kFLOAT) && (in.format == nvinfer1::TensorFormat::kLINEAR);
+  }
+
+  // pos == 1, accessing information about output tensor
+  const PluginTensorDesc& out = inOut[1];
+
+  return (in.type == out.type) && (in.format == out.format);
+}
+
+void AdaptiveMaxPool2dPlugin::configurePlugin(
+    const nvinfer1::DynamicPluginTensorDesc* in,
+    int nbInputs,
+    const nvinfer1::DynamicPluginTensorDesc* out,
+    int nbOutputs) {
+  dtype_ = DataType::kFLOAT;
+}
+
+size_t AdaptiveMaxPool2dPlugin::getWorkspaceSize(
+    const nvinfer1::PluginTensorDesc* inputs,
+    int nbInputs,
+    const nvinfer1::PluginTensorDesc* outputs,
+    int nbOutputs) const {
+  return 0;
+}
+
+int AdaptiveMaxPool2dPlugin::enqueue(
+    const nvinfer1::PluginTensorDesc* inputDesc,
+    const nvinfer1::PluginTensorDesc* outputDesc,
+    const void* const* inputs,
+    void* const* outputs,
+    void* workspace,
+    cudaStream_t stream) {
+#if NV_TENSORRT_MAJOR < 7 || (NV_TENSORRT_MAJOR == 7 && NV_TENSORRT_MINOR < 1)
+  at::Tensor input = at::from_blob((void*)inputs[0], util::toVec(inputDesc->dims), [](void*) {}, tensor_options_);
+  at::Tensor output = at::from_blob(
+      outputs[0], util::volume(outputDesc->dims), [](void*) {}, tensor_options_);
+  at::Tensor indices = at::from_blob(
+      outputs[1], util::volume(outputDesc->dims), [](void*) {}, index_tensor_options_);
+
+  at::cuda::CUDAStream torch_stream = at::cuda::getStreamFromPool();
+  at::cuda::CUDAStreamGuard torch_guard(torch_stream);
+
+  cudaEvent_t event;
+  cudaEventCreate(&event);
+  cudaEventRecord(event, stream);
+
+  cudaStreamWaitEvent(torch_stream.stream(), event, 0);
+
+  if (mode_ == "adaptive_max_pool2d") {
+    at::adaptive_max_pool2d_out(output, indices, input, {size_[0], size_[1]});
+  }
+
+  cudaEvent_t torch_event;
+  cudaEventCreate(&torch_event);
+  cudaEventRecord(torch_event, torch_stream.stream());
+
+  cudaStreamWaitEvent(stream, torch_event, 0);
+
+  cudaEventDestroy(event);
+  cudaEventDestroy(torch_event);
+
+  return 0;
+#else

```

Comment:
Since the pytorch and trt are updated, the original apative_avg_pool2d doesn't work either, my adaptive_max_pool2d was wriiten refered to adpative_avg_pool2d plugin.

Reply:
Can you elaborate what you meant by this statement ? 
`Since the pytorch and trt are updated, the original apative_avg_pool2d doesn't work either` 
The tests seem to be passing for me with Pyt 1.7.1 and cudnn8.

Reply:
In core/conversion/converters/impl/plugins/interpolate_plugin.cpp: 191
```cpp
#if NV_TENSORRT_MAJOR < 7 || (NV_TENSORRT_MAJOR == 7 && NV_TENSORRT_MINOR < 1)
  at::Tensor input = at::from_blob((void*)inputs[0], util::toVec(inputDesc->dims), [](void*) {}, tensor_options_);
  at::Tensor output = at::from_blob(
      outputs[0], util::volume(outputDesc->dims), [](void*) {}, tensor_options_);

  at::cuda::CUDAStream torch_stream = at::cuda::getStreamFromPool();
  at::cuda::CUDAStreamGuard torch_guard(torch_stream);

  cudaEvent_t event;
  cudaEventCreate(&event);
  cudaEventRecord(event, stream);

  cudaStreamWaitEvent(torch_stream.stream(), event, 0);

  if (mode_ == "linear") {
    at::upsample_linear1d_out(output, input, {size_[0]}, align_corners_);
  } else if (mode_ == "bilinear") {
    at::upsample_bilinear2d_out(output, input, {size_[0], size_[1]}, align_corners_);
  } else if (mode_ == "trilinear") {
    at::upsample_trilinear3d_out(output, input, {size_[0], size_[1], size_[2]}, align_corners_);
  } else if (mode_ == "adaptive_pool2d") {
    at::adaptive_avg_pool2d_out(output, input, {size_[0], size_[1]});
  }

  cudaEvent_t torch_event;
  cudaEventCreate(&torch_event);
  cudaEventRecord(torch_event, torch_stream.stream());

  cudaStreamWaitEvent(stream, torch_event, 0);

  cudaEventDestroy(event);
  cudaEventDestroy(torch_event);

  return 0;
#else
  // TODO: When PyTorch updates to cuDNN 8 try moving back to CUDA based ATen
  // kernels HACK: WAR because there is a segfault if you try to create a CUDA
  // Tensor in the context of TensorRT execution
  float* input_blob = (float*)malloc(util::volume(inputDesc->dims) * sizeof(float));
  cudaMemcpyAsync(
      input_blob,
      static_cast<const void*>(inputs[0]),
      util::volume(inputDesc->dims) * sizeof(float),
      cudaMemcpyDeviceToHost,
      stream);
  cudaStreamSynchronize(stream);

  at::Tensor input = at::from_blob((void*)input_blob, util::toVec(inputDesc->dims), tensor_options_);

  at::Tensor output;
  if (mode_ == "adaptive_pool2d") {
    output = at::adaptive_avg_pool2d(input, {size_[0], size_[1]});
  }

  cudaMemcpyAsync(
      outputs[0], output.data_ptr(), util::volume(outputDesc->dims) * sizeof(float), cudaMemcpyHostToDevice, stream);
  cudaStreamSynchronize(stream);

  free(input_blob);

  return 0;
#endif
```
The cuda version can run successfully througn TensorRT 7.0, cudnn 7.6.5. But when I update TensorRT to 7.2.1 and cudnn to 8.0, the cuda version plugin failed in testing, and only the cpu version works.

You can test the cuda version by change
```
#if NV_TENSORRT_MAJOR < 7 || (NV_TENSORRT_MAJOR == 7 && NV_TENSORRT_MINOR < 1)
```
to 
```
#if NV_TENSORRT_MAJOR < 7 || (NV_TENSORRT_MAJOR == 7)
```